main.go

```diff
@@ -475,22 +475,93 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
 	return status, nil
 }
 
-func processSecret(kubeClient *k8s.Client, secret *corev1.Secret, initiator string) (status string, err error) {
+func isEventExist(kubeClient *k8s.Client, namespace string, name string, event *corev1.Event ) (exist bool,err error){
+	err = kubeClient.Get(context.Background(), namespace, name, event)
+	exist = true
```

Comment:
You can inline this variable by returning `false` or `true` in the return statements.

Reply:
I have now three options (found, not found or error)